docs/5.Minting-Module.md

```diff
@@ -0,0 +1,693 @@
+
+The control parameter will be calculated in the Rebalancer module.  The value of the control parameter will be a decimal in the range of 0 to 1 and will modify the **change** to the current inflation rate. This means that depending on the circulation of supplies of tokens in the Validator and Liquidity Pool subsystems, the parameter will **maintain** the Cosmos calculations for inflation rate change with a value of **1**, **remove** the Cosmos calculations for inflation rate change with a value of **0**, or **dampen** the Cosmus calculations for inflation rate change with any value **between 0 and 1**.
+
+## Cosmos-SDK Wrapped Mint Module
+
+In order to wrap the Cosmos-SDK mint module, we create a new module, import the Cosmos-SDK mint module, redefine the functions that need to be changed, and import this new module into Sifchain.
+
+![](images/MintModule/mint.png)
+
+The wrapped module will have the following structure:
+
+    -x/
+        -mint/
+            -internal/types/
+                -minter.go
+                -expected_keepers.go
+				-params.go
+			-abci.go
+            -alias.go
+            -cosmos_alias.go
+            -module.go
+
+
+**minter.go** The minter type is where we use our Rebalancing control parameter in the NextInflationRate() function. Everything else is unchanged from Cosmos' mint module.
+
+```golang
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Minter represents the minting state.
+type Minter struct {
+	Inflation        sdk.Dec `json:"inflation" yaml:"inflation"`                 // current annual inflation rate
+	AnnualProvisions sdk.Dec `json:"annual_provisions" yaml:"annual_provisions"` // current annual expected provisions
+}
+
+// NewMinter returns a new Minter object with the given inflation and annual
+// provisions values.
+func NewMinter(inflation, annualProvisions sdk.Dec) Minter {
+	return Minter{
+		Inflation:        inflation,
+		AnnualProvisions: annualProvisions,
+	}
+}
+
+// InitialMinter returns an initial Minter object with a given inflation value.
+func InitialMinter(inflation sdk.Dec) Minter {
+	return NewMinter(
+		inflation,
+		sdk.NewDec(0),
+	)
+}
+
+// DefaultInitialMinter returns a default initial Minter object for a new chain
+// which uses an inflation rate of 13%.
+func DefaultInitialMinter() Minter {
+	return InitialMinter(
+		sdk.NewDecWithPrec(13, 2),
+	)
+}
+
+// validate minter
+func ValidateMinter(minter Minter) error {
+	if minter.Inflation.IsNegative() {
+		return fmt.Errorf("mint parameter Inflation should be positive, is %s",
+			minter.Inflation.String())
+	}
+	return nil
+}
+
+// NextInflationRate returns the new inflation rate for the next hour.
```

Comment:
Would we use block time ,or number of blocks to keep a track of time ?

Reply:
This comment is from the cosmos mint module and isn't even accurate, I'll remove it

Reply:
Nextinflationrate is calculated every block, not every hour